tools/ui-components/src/toggle-button/toggle-button.tsx

```diff
@@ -0,0 +1,119 @@
+
+const defaultClassNames = [
+  'relative',
+  'border-3',
+  'text-center',
+  'inline-block',
+  'active:before:w-full',
+  'active:before:h-full',
+  'active:before:absolute',
+  'active:before:inset-0',
+  'active:before:border-3',
+  'active:before:border-transparent',
+  'active:before:bg-gray-900',
+  'active:before:opacity-20',
+  'focus:outline-none', // Hide the default browser outline
+  'focus:ring',
+  'focus:ring-focus-outline-color',
+  'ui-checked:cursor-not-allowed', // ui-checked from @headlessui/tailwindcss plugin
+  'aria-disabled:cursor-not-allowed',
+  'aria-disabled:opacity-50'
+];
+
+const computeClassNames = ({
+  size,
+  variant,
+  disabled
+}: {
+  size: ButtonSize;
+  variant: ButtonVariant;
+  disabled?: boolean;
+}) => {
+  const classNames = [...defaultClassNames];
+
+  switch (variant) {
+    case 'danger':
+      classNames.push(
+        'border-foreground-danger',
+        'bg-background-danger',
+        'text-foreground-danger',
+        'ui-checked:bg-foreground-danger',
+        'ui-checked:text-background-danger',
+        ...(disabled
+          ? ['active:before:hidden']
+          : [
+              'hover:bg-foreground-danger',
+              'hover:text-background-danger',
+              'dark:hover:bg-background-danger',
+              'dark:hover:text-foreground-danger',
+              'ui-checked:hover:bg-background-danger',
+              'ui-checked:hover:text-foreground-danger'
+            ])
+      );
+      break;
+    // default variant is 'primary'
+    default:
+      classNames.push(
+        'border-foreground-secondary',
+        'bg-background-quaternary',
+        'text-foreground-secondary',
+        'ui-checked:bg-foreground-primary',
+        'ui-checked:text-background-primary',
+        ...(disabled
+          ? ['active:before:hidden']
+          : [
+              'hover:bg-foreground-primary',
+              'hover:text-background-primary',
+              'dark:hover:bg-background-primary',
+              'dark:hover:text-foreground-primary',
+              'ui-checked:hover:bg-background-quaternary',
+              'ui-checked:hover:text-foreground-secondary'
+            ])
+      );
+  }
+
+  switch (size) {
+    case 'large':
+      classNames.push('px-8 py-2.5 text-lg');
+      break;
+    case 'medium':
+      classNames.push('px-6 py-1.5 text-md');
+      break;
+    // default size is 'small'
+    default:
+      classNames.push('px-5 py-1 text-sm');
+  }
+
+  return classNames.join(' ');
+};
+
+export const ToggleButton = ({
+  size = 'small',
+  variant = 'primary',
+  disabled,
+  children,
+  checked,
+  onChange
+}: ToggleButtonProps) => {
+  const classNames = computeClassNames({ size, variant, disabled });
+
+  const handleChange = () => {
+    if (!disabled && onChange) {
+      onChange(true);
+    }
+  };
+
+  return (
+    <Switch
+      checked={checked}
+      onChange={handleChange}
+      className={classNames}
+      aria-disabled={disabled}
```

Comment:
while testing, I found out, if you change this to `disabled`, it will break the component

![error about importing the component wrongfully](https://user-images.githubusercontent.com/88248797/207539870-52956309-565a-463c-bc4d-a8a6a963ede2.png)

```ts
      disabled={disabled}
```

---

If you faced this and fixed, nicely done. It would be a nightmare to solve